nixos/modules/services/web-servers/minio.nix

```diff
@@ -29,23 +35,18 @@ in
       description = "The config directory, for the access keys and other settings.";
     };
 
-    accessKey = mkOption {
-      default = "";
+    rootUser = mkOption {
       type = types.str;
       description = ''
-        Access key of 5 to 20 characters in length that clients use to access the server.
-        This overrides the access key that is generated by minio on first startup and stored inside the
-        <literal>configDir</literal> directory.
+        The access key for the root user.
       '';
     };
 
-    secretKey = mkOption {
-      default = "";
+    rootPassword = mkOption {
```

Comment:
Can we kick this PR up a notch and remove secrets from the nix store?

Reply:
Sure if somebody can point me to how to do this?

Reply:
I think [these](https://github.com/NixOS/nixpkgs/blob/d42cd445dde587e9a993cd9434cb43da07c4c5de/nixos/modules/services/security/oauth2_proxy.nix#L550) [two](https://github.com/NixOS/nixpkgs/blob/7968281008dfa46f83e213a3a64eca65562086b2/nixos/modules/services/web-apps/miniflux.nix#L54) descriptions work well when combined with `EnvironmentFile`. Does that seem like a good solution to you too?

Reply:
Update the MR to use `rootCredentialsFile`